scipy/linalg/interpolative.py

```diff
@@ -424,65 +433,45 @@ def _is_real(A):
-        If 'default', the random seed is reset to a default value.
-
-        If `seed` is a sequence containing 55 floating-point numbers
-        in range [0,1], these are used to set the internal state of
-        the generator.
-
-        If the value is an integer, the internal state is obtained
-        from `numpy.random.RandomState` (MT19937) with the integer
-        used as the initial seed.
-
-        If `seed` is omitted (None), ``numpy.random.rand`` is used to
-        initialize the generator.
+    This function used to set the seed of the randomization algorithms used in the
+    `scipy.linalg.interpolative` functions written in Fortran77.
 
+    The library has been ported to Python and now the functions use the native NumPy
+    generators and this function has no content and returns None. Thus this function
+    should not be used and will be removed in SciPy version 1.17.0.
     """
-    # For details, see :func:`_backend.id_srand`, :func:`_backend.id_srandi`,
-    # and :func:`_backend.id_srando`.
-
-    if isinstance(seed, str) and seed == 'default':
-        _backend.id_srando()
-    elif hasattr(seed, '__len__'):
-        state = np.asfortranarray(seed, dtype=float)
-        if state.shape != (55,):
-            raise ValueError("invalid input size")
-        elif state.min() < 0 or state.max() > 1:
-            raise ValueError("values not in range [0,1]")
-        _backend.id_srandi(state)
-    elif seed is None:
-        _backend.id_srandi(np.random.rand(55))
-    else:
-        rnd = np.random.RandomState(seed)
-        _backend.id_srandi(rnd.rand(55))
+    warnings.warn("`scipy.linalg.interpolative.seed` is deprecated and will be "
+                  "removed in SciPy 1.17.0.", DeprecationWarning, stacklevel=2)
 
 
 def rand(*shape):
     """
-    Generate standard uniform pseudorandom numbers via a very efficient lagged
-    Fibonacci method.
+    This function used to set the seed of the randomization algorithms used in the
+    `scipy.linalg.interpolative` functions written in Fortran77.
 
-    This routine is used for all random number generation in this package and
-    can affect ID and SVD results.
+    The library has been ported to Python and now the functions use the native NumPy
+    generators. Thus this function should not be used and will be removed in the
+    SciPy version 1.17.0.
+
+    If pseudo-random numbers are needed, NumPy should be used instead:
+
+        >>> import numpy as np
+        >>> rng = np.random.default_rng()
+        >>> rng.uniform(low=0., high=1.0, size=shape)
 
     Parameters
     ----------
     *shape
         Shape of output array
 
     """
-    # For details, see :func:`_backend.id_srand`, and :func:`_backend.id_srando`.
-    return _backend.id_srand(np.prod(shape)).reshape(shape)
+    warnings.warn("`scipy.linalg.interpolative.rand` is deprecated and will be "
+                  "removed in SciPy 1.17.0.", DeprecationWarning, stacklevel=2)
```

Comment:
This is an unwanted function now. The stacklevel should be corrected and should be added to the deprecation tracker.